### ISSUE TITLE
Color reproduction problems

### ISSUE THREAD

**Issue by mikkelam** [uber/ios-snapshot-test-case#27]
I'm not really sure how to explain or even reproduce this problem, just hoping someone else has this problem..

Anyway, my issue is that there is a slight difference in color based on the machine my screenshot test is run on (in this case circleCI and my machine)

See the screenshots: first one is reference (my machine), next is the fail (circleCI) and the third is the diff

<img width="375" alt="reference_screenshot" src="https://user-images.githubusercontent.com/4174131/37923481-895ffa22-312f-11e8-8e9d-866c083f9726.png">
<img width="375" alt="failed_screenshot" src="https://user-images.githubusercontent.com/4174131/37923482-897ea968-312f-11e8-96d9-4e005772cd77.png">
<img width="375" alt="diff_screenshot" src="https://user-images.githubusercontent.com/4174131/37923483-899cedf6-312f-11e8-8db0-60c928966c7c.png">

Basically there is an extremely minor color difference on the blue button (took me a while to find)

The code basically puts the VC as rootVC and calls layoutifneeded on the view

**Comment by alanzeino** [comment]
Did you generate the snapshots using the same Base SDK as you are using in your simulators on CI to run the tests?

**Comment by dominikoledzki** [comment]
Do you use the same device in both environments? Pay attention especially to devices with vs. without True Tone display.

**Comment by mikkelam** [comment]
> Did you generate the snapshots using the same Base SDK as you are using in your simulators on CI to run the tests?

Yes same base SDK (11.2) with xcode 9.2 (9C40B).


> Do you use the same device in both environments? Pay attention especially to devices with vs. without True Tone display.

Both are iphone 6 as well

**Comment by alanzeino** [comment]
Honestly since I don't have experience with CircleCI, I can't say I've seen this before. 

But my first suggestion would be to reset simulators before you run tests using `xcrun`. That way, you can be 100% sure that the simulator you're using is 'pristine' for your test run. `xcodebuild`'s `destination` parameter can be pointed to a custom one–time simulator GUID if you use `xcrun simctl create` and then pass the returned GUID to destination using the `id` key.

Other general troubleshooting things would be to reset the simulator you're using locally to generate these images, also to run the test step using `xcodebuild` on the command line locally to see if the issue manifests with `xcodebuild` locally.
